controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go

```diff
@@ -341,3 +352,27 @@ func getAKOPackageRefFromClusterBootstrap(log logr.Logger, cb *runv1alpha3.Clust
 	}
 	return -1, nil
 }
+
+func validateADCAndClusterIpFamily(cluster *clusterv1.Cluster, adc *akoov1alpha1.AKODeploymentConfig, isVIPProvider bool, log logr.Logger) error {
+	adcIpFamily := "V4"
+	if adc.Spec.ExtraConfigs.IpFamily != "" {
+		adcIpFamily = adc.Spec.ExtraConfigs.IpFamily
+	}
+	clusterIpFamily, err := utils.GetClusterIPFamily(cluster)
+	if err != nil {
+		log.Error(err, "can't get cluster ip family")
+		return err
+	}
+	if (adcIpFamily == "V4" && clusterIpFamily == "V6") || (adcIpFamily == "V6" && clusterIpFamily == "V4") {
+		errInfo := "AKO with IP family " + adcIpFamily + " can not work together with cluster with IP family " + clusterIpFamily
+		return errors.New(errInfo)
+	}
+	if isVIPProvider {
```

Comment:
Could you add some comments in this function to make it easier to understand?

Such as why it can't work together, AKO's limitations etc. 

We might understand this quite well now, but we will forget in future

Reply:
sounds good